text/0232-maturity-of-otel.md

```diff
@@ -0,0 +1,153 @@
+
+Only once we reach the "stable" level at the project level will we be ready to consider graduating OpenTelemetry at the CNCF, with all of the Tier-1 SIGs being in the context of the graduation, including their Tier-1 Components. This OTEP provides a framework for describing project maturity as a whole.
+
+## Glossary
+
+* *SIG:* an official SIG within the OpenTelemetry project, like the Collector or Java. Given that some Working Groups (WGs) are effectively SIGs, like the Semantic Conventions WG, WGs will be accepted on an individual basis.
+* *Tier-1 SIG:* a SIG within the OpenTelemetry project deemed critical by the GC/TC.
+* *Candidate Tier-1 SIG:* a SIG that is deemed as critical for the future of the project, but might not be of the same maturity level yet. It's expected to reach that level eventually, and other Tier-1 SIGs should be prepared to accommodate the new component as needed.
+* *Component:* module maintained by a SIG like a Collector exporter or an SDK sampler.
+* *Tier-1 Component:* a component selected by project maintainers as critical for the SIG. For instance, the OTLP receiver will likely be selected as a Tier-1 Component of the Collector. See Tier-1 Components for more details.
+* *Candidate Tier-1 Component:* similar to "Candidate Tier-1 SIG", but for a component within a SIG.
+* *Dependency:* an OpenTelemetry SIG or component that is required by another SIG or component. For instance, the OTLP Logs Data Model from the Spec is required for the Collector's Logging support.  
+
+## Explanation
+
+### Maturity levels
+
+SIGs and components of a project MUST have a declared maturity level established by SIG maintainers (SIGs). The maturity of the SIG is assessed by its maintainers, while the maturity of the components might be delegated by the maintainers to code owners. The maturity level for a SIG or component is, at most, the lowest level of its dependencies or priority components. For instance, if the Collector SIG maintainers declare the "otlpreceiver" component as a tier-1 component of the Collector core distribution, and the "otlpreceiver" declares a dependency on the OpenTelemetry Collector API "config" package which is marked with a maturity level of "beta", the "otlpreceiver" can be at most "beta". The OpenTelemetry Collector "core" distribution is also affected so that it can be at most "beta".
+
+Only once all dependencies are marked as "stable" MAY a component be marked as "stable". Only once all Tier-1 Components of a SIG are "stable" MAY that SIG be marked as "stable".
```

Comment:
A SiG is a group of people working a component, a component *itself* is the software delivered by those people.

I'm not sure we should conflate the two.  e.g. I'm fine if semantic conventions are considered a critical / tier-1 component, but I'm not sure the *working group* is.

Maybe it would make sense to focus this proposal on components?   Rather than talking about SiGs, we can focus on OWNERS/MAINTAINERS of those components.

Reply:
The idea is that SIG maintainers would have the power to determine which deliverables are included in the SIG's tier-1 list. For instance, the Collector might include the "core" distribution as "tier-1" but not the "contrib" distribution.

Reply:
In the context of this OTEP, we have a list of tier-1 SIGs, each one delivering components of different stability levels. Once all components for a given SIG are stable, that SIG is declared stable. Once all tier-1 SIGs are stable, can we call OpenTelemetry stable.

Reading that paragraph without this context makes it sound weird: there's no attempt to mark SIGs stable or not, it's just a way to see how far we are into the overall stability of the project, not a judgment of the SIG itself.